server/client/js/main.js

```diff
@@ -6,64 +6,158 @@ let users = db.collection('users');
 let rooms = db.collection('rooms');
 
 let socket;
+let userDoc;
+let lastOperation;
 
+// Once DOM is ready
+$(() => {
+    // Sign in logic
+    $("#signInForm").submit(e => {
+        e.preventDefault();
+        let email = $("#signInEmail").val();
+        let password = $("#signInPassword").val();
+        auth.signInWithEmailAndPassword(email, password)
+            .catch((error) => {
+                switch (error.code) {
+                    case "auth/invalid-email":
+                        alert("Invalid email.");
+                        break;
+                    case "auth/user-disabled":
+                        alert("User disabled.");
+                        break;
+                    case "auth/user-not-found":
+                        $("#signUpEmail").val(email);
+                        $("#signUpPassword").val(password);
+                        $("#signIn").hide();
+                        $("#signUp").show();
+                        break;
+                    case "auth/wrong-password":
+                        alert("Incorrect password.");
+                        break;
+                    default:
+                        alert(error.message);
+                }
+            });
+    });
```

Comment:
Sign-in function, allows for validation/UI logic on sign-in.

Reply:
Does firebase force users to choose strong passwords?

Im a little worried of having too much friction to start messing around in the installation. 

How bad would it be to just ask people for a display name and that’s it? Would that make it too easy to make a bunch of fake names to bypass the rules?

Reply:
It does require some strength, but it's not much -- just needs to be 6 characters or more.

We could potentially ask for just a display name, would need a bit more work but it's doable if we want to go with that.